qutebrowser/utils/urlmatch.py

```diff
@@ -177,6 +179,15 @@ def _init_host(self, parsed):
             assert self._host is None
             return
 
+        if parsed.netloc.startswith('['):
+            # Using QUrl parsing to minimize ipv6 addresses
+            url = QUrl()
+            url.setHost("[" + parsed.hostname + "]")
+            if url.host() == "":
+                raise ParseError("Invalid IPv6 URL"+parsed.hostname)
```

Comment:
Any reason you aren't using `if url.isValid():` and `url.errorString()` here like I said in my earlier comment? :wink:

Reply:
no, I just overlooked it